src/index.ts

```diff
@@ -89,28 +27,21 @@ export = class OutputDcatUs11 {
     res.set('Content-Type', 'application/json');
 
     try {
-      const hostname = req.hostname;
-      const siteModel = await this.fetchSite(hostname, this.getRequestOptions(portalUrl));
-
-      // Use dcatConfig query param if provided, else default to site's config
-      let dcatConfig = typeof req.query.dcatConfig === 'string'
-        ? this.parseProvidedDcatConfig(req.query.dcatConfig as string)
-        : req.query.dcatConfig;
-
-      if (!dcatConfig) {
-        dcatConfig = _.get(siteModel, 'data.feeds.dcatUS11');
+      const { res: { locals: { feedTemplate, feedTemplateTransforms, nonEditableTemplateFields } } }: {
+        res?: {
+          locals?: {
+            feedTemplate?: any,
+            feedTemplateTransforms?: TransformsList,
+            nonEditableTemplateFields?: string[]
+          }
+        }
+      } = req;
+
+      if (!feedTemplate) {
+        throw Error('Feed template not provided!');
       }
 
-      // TODO: We only pass in hostname because some site item urls are out of sync, causing invalid urls for
-      // landingPage and identifier. If we can resolve the syncing issues, we can omit hostname and just use
-      // the absolute url we get from getContentSiteUrls()
-      const { stream: dcatStream, dependencies } = getDataStreamDcatUs11(hostname, siteModel, dcatConfig);
-
-      const apiTerms = await getApiTermsFromDependencies(dependencies);
-
-      // Request a single dataset if id is provided, else default to site's catalog
-      const id = String(req.query.id || '');
-      req.res.locals.searchRequest = this.getDatasetSearchRequest(id, apiTerms) || this.getCatalogSearchRequest(req, _.get(siteModel, 'data.catalog'), apiTerms);
+      const { stream: dcatStream } = getDataStreamDcatUs11(feedTemplate, feedTemplateTransforms, nonEditableTemplateFields);
```

Comment:
Seems like this will fail if the user's Koop instance doesn't add all these things to `locals`.  Are these parameters things that could conceivable arrive on the req.query or req.body? Also are there default we can use in the event the request doesn't have all of these? If not, it would seem that we would want to a 400 error.

Reply:
> Are these parameters things that could conceivable arrive on the req.query or req.body? 

Currently, we expect users using the plugin to have a feed template object in their koop instance but we can allow template to be passed via req.query or req.body but should remove existing template parsing via query in hub feeds api as it would be redundant.

>Also are there default we can use in the event the request doesn't have all of these? I

We could have a default one but dcat us feed has properties like title, landingPage whose values are dynamic. Feed template is essential to map these keys to appropriate values from datasource.  There is no common data source as providers could be different. so, common mapping is not possible.

>it would seem that we would want to a 400 error.

Currently responds if 400 error if `feedTemplate` not set.